### DISCUSSION TITLE
_tvp constraints for MPC

### DISCUSSION THREAD

**Discussion by shantropy** [do-mpc/do-mpc#208] [Q&A] [answered]
Is it possible to impose _tvp constraints for MPC? e.g., In the [DIP example](https://github.com/do-mpc/do-mpc/blob/v4.1.0/examples/DIP/template_mpc.py) obstacle location is a function of time

So far, I've had no luck making it work. I did manage to define a time varying obstacle but solver does not recognize the obstacle movement when solving for the _optimal path_ at each iteration.

**Comment by 4flixt** [comment, marked as answer]
Hi @samanmostafavi ,

you can absolutely use time-varying parameters in your constraint formulation. You can simply use your introduced ``_tvp`` variables in [set_nl_cons](https://www.do-mpc.com/en/latest/api/do_mpc.controller.MPC.set_nl_cons.html). Unfortunately, we have no example available at the moment where this is implemented. 

Best
4flixt
